Netable/Netable/Netable.swift

```diff
@@ -170,13 +189,38 @@ public actor Netable {
         return (task: task, subject: resultSubject.receive(on: RunLoop.main))
     }
 
-    private func startRequestTask<T: Request>(_ request: T, urlRequest: URLRequest, id: String) async throws -> T.FinalResource {
+    private func startRequestTask<T: Request>(
+        _ request: T,
+        interceptor: Interceptor?,
```

Comment:
What do you see as the main use case for per request interceptors? It seems theoretically useful, but I'm having a hard time coming up with a concrete use case. It seems like any time you are passing in a custom interceptor with a request, you could just as easily have the code that is calling it just switch between the real call and whatever it is faking at the call point without much more code complexity than building the interceptor.

Reply:
I think you've pretty much got it. I'm a little concerned that you could end up in a situation where you've got one global interceptor that's just a giant switch statement and thought it might be a little cleaner to just have per request interceptors in that case, or maybe you've got a couple different things you're trying to modify for different requests (some need different headers, some need mocked data, some need an endpoint swap) and those needs overlap in weird ways?

I'm not super attached to this, I think we could probably just simplify things here and say you can only attach interceptors globally if you feel strongly about it, I just thought it might be useful.

Reply:
You can always do them as multiple `RequestInterceptor`s that are applied globally and just pattern matching the URL though, something like:

```
class InterceptorA: RequestInterceptor {
   func apply(...) {
       if request.url.path.contains("/somePath") {
           return URLRequest(...)
       }

       return request
}

class InterceptorB: RequestInterceptor {
   func apply(...) {
       if request.url.path.contains("/someOtherPath") {
           return URLRequest(...)
       }

       return request
}

config.interceptors = [InterceptorA(), InterceptorB()]
```

This seems like a better style to me, since you don't need to know in most of the networking code which calls might be affected by which mocks. Per request interceptors seem like they are only useful if you are making the same requests from multiple locations and only want an interceptor to apply to SOME of them, which I can't actually think if a great use case for.

Reply:
Yeah you're right, I'm on board with just removing request level interceptors. Updated.